internal/librarian/pullrequest_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package librarian
+
+import (
+	"errors"
+	"slices"
+	"strings"
+	"testing"
+)
+
+func TestAddErrorToPullRequest(t *testing.T) {
+	pr := PullRequestContent{
+		Successes: []string{"s1", "s2"},
+		Errors:    []string{"e1", "e2"},
+	}
+	addErrorToPullRequest(&pr, "library-id", errors.New("bang"), "generating")
+	if !slices.Equal([]string{"s1", "s2"}, pr.Successes) {
```

Comment:
not required here by any means but if you want to compare the whole struct to another struct with nice diffs a lot of our Go libraries use: https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff

Reply:
Will leave this one for now, but we can chat about idiomatic struct testing when we discuss testing in general.

Reply:
ack